### ISSUE TITLE
What templates should we provide?

### ISSUE THREAD

**Issue by cknitt** [rescript-lang/create-rescript-app#11]
Currently (as of v1.4.0) providing these two:
* Basic: "https://github.com/rescript-lang/rescript-project-template.git"
* Next.js: "https://github.com/ryyppy/rescript-nextjs-template.git"

What other templates could/should we provide (while not going overboard with this and still keeping things minimalistic)?

How/where should the templates be stored to enforce consistency and avoid breakage?

Should we provide "experimental" templates for trying out things like JSX4?

**Comment by DZakh** [comment]
I'd like to have a script that would integrate ReScript into an existing project.
- Install rescript
- Add bs-config
- Add npm scripts
- Add internal files to `.gitignore`

**Comment by cknitt** [comment]
Yes, that would be nice. Maybe we could add that as an option in create-rescript-app.

Although people probably wouldn't expect this / wouldn't be looking for this functionality in a create-xxx package? 🤔

**Comment by DZakh** [comment]
Yeah, probably, more obvious to have such behavior in the `rescript init`

**Comment by zth** [comment]
Things that come to mind:
* A vite-based template
* Remix
* Node + Express

**Comment by DZakh** [comment]
I'm writing an article how to use ReScript for a CLI. Can be packed in a template repo.
Also, I've started working on [platformatic](https://oss.platformatic.dev/) bindings, I'd prefer having a template with it or pure Fastify over the one with express.

**Comment by DZakh** [comment]
@cknitt What do you think of `rescript-init` npm package? I can create it after finishing working on the https://github.com/DZakh/rescript-stdlib-cli that I described in the https://forum.rescript-lang.org/t/wip-the-ultimate-answer-to-belt-vs-js-in-rescript/3844. They should have pretty similar functionality, so wouldn't take much effort.
I'm planning to start writing articles about ReScript and it would be nice to simplify the step of configuring the project 😊

**Comment by cknitt** [comment]
@zth I would also like to have a Vite template.

There is https://github.com/jihchi/vitejs-template-react-rescript, but that one already includes a bit too much stuff IMHO (rescript-webapi, rescript-vitest).

I would like to have a very simple template with only Vite - or maybe Vite + Tailwind CSS, that's a very nice combo that we are using in our projects.

**Comment by cknitt** [comment]
Regarding express vs. fastify vs. platformatic: I am currently seeing the following weekly download numbers on npm:

* express: 26.957.203
* fastify: 759.168
* platformatic: 77

I think we should provide a template for express first because it still seems to be the most popular web framework for Node.js. Having a fastify template in addition might also be nice. I would not do one for platformatic.

**Comment by fhammerschmidt** [comment]
I would welcome a "rescript-development-testing" template that somehow comes with the latest master or at least the `next` npm package (currently [10.1.0-rc.2](https://www.npmjs.com/package/rescript/v/10.1.0-rc.2)) to lower the barrier for others to report bugs and test new features.

**Comment by cknitt** [comment]
> @cknitt What do you think of `rescript-init` npm package? ...

@DZakh Sorry for the late reply! Yes, it will definitely be nice to have a means to integrate ReScript into an existing project like you described. I am still a bit unsure about where we'd best put that functionality, but it will surely be helpful if you get the ball rolling with a separate `rescript-init` package. 👍

**Comment by Frank-III** [comment]
I think we should introduce a React Router template now that it’s a Vite plugin and the ReScript site is moving to React Router: https://github.com/rescript-lang/rescript-lang.org/pull/1115

**Comment by jderochervlk** [comment]
I would love to do this!
